ocaml/xapi/xapi_sr.ml

```diff
@@ -61,30 +61,10 @@ let scan_finished sr =
       Condition.broadcast scans_in_progress_c
   )
 
-module Throttle () = struct
-  let semaphore = ref None
-
-  let m = Mutex.create ()
-
-  let get_semaphore () =
-    with_lock m (fun () ->
-        (* overridable on startup from config file, have
-         * to delay initialization *)
-        match !semaphore with
-        | None ->
-            let result = Semaphore.create !Xapi_globs.max_active_sr_scans in
-            semaphore := Some result ;
-            result
-        | Some s ->
-            s
-    )
-
-  let execute f = Semaphore.execute (get_semaphore ()) f
-end
-
-module AutoScanThrottle = Throttle ()
+module Size = struct let n = !Xapi_globs.max_active_sr_scans end
```

Comment:
This will evaluate Xapi_globs.max_active_sr_scans too early, before the config file or cmdline is parsed (it'll run as part of module startup code).
I think we need to delay evaluation of this by making it `unit -> int`.